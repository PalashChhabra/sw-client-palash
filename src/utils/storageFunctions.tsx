const ROOT_FAV_KEY = 'FAV_RESOURCE';

// get Favourite Resources
export const getFavs = (): string[] => {
	const key = ROOT_FAV_KEY;

	let favs = localStorage.getItem(key) || '[]';

	try {
		favs = JSON.parse(favs);
		if (!favs) return [];
	} catch {
		localStorage.removeItem(ROOT_FAV_KEY);
	}

	// @ts-ignore
	return favs;
};

// toggle action of favourite
export const toggleFav = (item: string): void => {
	const favs = getFavs();
	const key = ROOT_FAV_KEY;

	if (favs.includes(item))
		localStorage.setItem(key, JSON.stringify(favs.filter(f => f !== item)));
	else localStorage.setItem(key, JSON.stringify([...favs, item]));
};
