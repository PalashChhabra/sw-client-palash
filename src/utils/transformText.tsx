// to transform text and remove underscore
export const transformText = (str: string): string =>
	(str.length > 1 ? str[0].toUpperCase() + str.slice(1).replace(/_/g, ' ') : str);
