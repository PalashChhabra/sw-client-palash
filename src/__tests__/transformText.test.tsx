/* eslint-disable camelcase */
import { transformText } from '../utils/transformText';

const resourceData = [
	{
		originalKey: 'orbital_period',
		expectedKey: 'Orbital period',
	},
	{
		originalKey: 'surface_water',
		expectedKey: 'Surface water',
	},
	{
		originalKey: 'rotation_period',
		expectedKey: 'Rotation period',
	},
];

describe('should tranform data Keys and replace underscores', () => {
	for (const { originalKey, expectedKey } of resourceData) {
		it(`expect (${originalKey} to match ${expectedKey}`, () => {
			const output = transformText(originalKey);

			expect(output).toEqual(expectedKey);
			expect(output).toMatchSnapshot();
		});
	}
});
