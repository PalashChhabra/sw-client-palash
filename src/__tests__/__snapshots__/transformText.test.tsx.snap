// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should tranform data Keys and replace underscores expect (orbital_period to match Orbital period 1`] = `"Orbital period"`;

exports[`should tranform data Keys and replace underscores expect (rotation_period to match Rotation period 1`] = `"Rotation period"`;

exports[`should tranform data Keys and replace underscores expect (surface_water to match Surface water 1`] = `"Surface water"`;
